include/exadg/postprocessor/pointwise_output_generator_base.h

```diff
@@ -0,0 +1,198 @@
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#ifndef INCLUDE_COMPRESSIBLE_NAVIER_STOKES_POSTPROCESSOR_POINTWISE_OUTPUT_GENERATOR_BASE_H_
+#define INCLUDE_COMPRESSIBLE_NAVIER_STOKES_POSTPROCESSOR_POINTWISE_OUTPUT_GENERATOR_BASE_H_
+
+#include <deal.II/base/conditional_ostream.h>
+#include <deal.II/base/hdf5.h>
+#include <deal.II/base/mpi_remote_point_evaluation.h>
+#include <deal.II/base/point.h>
+
+#include <deal.II/lac/la_parallel_vector.h>
+
+#include <deal.II/distributed/tria.h>
+
+#include <deal.II/fe/mapping.h>
+
+#include <deal.II/numerics/vector_tools.h>
+
+namespace ExaDG
+{
+template<int dim>
+struct PointwiseOutputDataBase
+{
+  using point_value_type = typename dealii::Point<dim>::value_type;
+
+  PointwiseOutputDataBase();
+
+  bool        write_output;
+  std::string directory;
+  std::string filename;
+
+  bool update_points_before_evaluation;
+
+  double start_time;
+  double end_time;
+  double interval_time;
+
+  std::vector<dealii::Point<dim>> evaluation_points;
+
+  void
+  print(dealii::ConditionalOStream & pcout) const;
+};
+
+template<int dim, typename Number>
+class PointwiseOutputGeneratorBase
+{
+public:
+  using VectorType       = dealii::LinearAlgebra::distributed::Vector<Number>;
+  using point_value_type = typename PointwiseOutputDataBase<dim>::point_value_type;
+
+  PointwiseOutputGeneratorBase(MPI_Comm const & comm);
+
+  virtual ~PointwiseOutputGeneratorBase() = default;
+
+  void
+  setup_base(dealii::DoFHandler<dim> const &      dof_handler_in,
+             dealii::Mapping<dim> const &         mapping_in,
+             PointwiseOutputDataBase<dim> const & pointwise_output_data_in);
+
+  void
+  evaluate(VectorType const & solution, double const & time, int const & time_step_number);
+
+  void
+  add_scalar_quantity_dataset(std::string const & name);
+
+  void
+  add_vector_quantity_datasets(std::string const & name);
+
+  void
+  write_scalar_quantity(std::string const & name, dealii::Vector<Number> const & hyperslab_data);
+
+  void
+  write_vectorial_quantity(std::string const & name, dealii::Vector<Number> const & hyperslab_data);
+
+  void
+  write_scalar_quantity(std::string const &         name,
+                        std::vector<Number> const & values,
+                        unsigned int const          first_component = 0)
+  {
+    AssertThrow(first_component == 0,
+                dealii::ExcMessage("This function can only be called with first_component==0."));
+
+    hyperslab_data = std::move(dealii::Vector<Number>(values.begin(), values.end()));
+    write_hyperslab(name, hyperslab_data);
+  }
+
+  template<int n_components>
+  void
+  write_scalar_quantity(std::string const &                                          name,
```

Comment:
I don't understand this function name in combination with `Tensor<1, n_components, Number>`? This raises the question whether we should distinguish at all between scalar/vectorial regarding the function names.

Reply:
Say we have a solution vector containing pressure and velocity, then we have to pick out the pressure or velocity before we write to the file. So basically we give the functions all results and leave it up to the function to extract the correct data and write it to the file. I would like to keep it this way if you don't have any concerns.

Reply:
see discussion in other thread.